esmvaltool/cmorizers/data/formatters/datasets/merra.ncl

```diff
@@ -0,0 +1,306 @@
+                 "tavgM_2d_rad_Nx.",                     \
+                 "tavgM_2d_chm_Fx.", "tavgM_2d_chm_Fx."/)   ; 2d chm
+
+  ; dataset doi numbers
+  DOI = (/"10.5067/YX0AVASQRTNW", "10.5067/YX0AVASQRTNW",  \  ; 3d asm
+          "10.5067/YX0AVASQRTNW", "10.5067/YX0AVASQRTNW",  \
+          "10.5067/YX0AVASQRTNW", "10.5067/YX0AVASQRTNW",  \
+          "10.5067/YX0AVASQRTNW", "10.5067/YX0AVASQRTNW",  \
+          "10.5067/YX0AVASQRTNW", "10.5067/YX0AVASQRTNW",  \
+          "10.5067/YX0AVASQRTNW",                          \
+          "10.5067/QL0PGBK2CYJS", "10.5067/QL0PGBK2CYJS",  \  ; 2d int
+          "10.5067/QL0PGBK2CYJS",                          \
+          "10.5067/6UX3EDUNVUFK", "10.5067/6UX3EDUNVUFK",  \  ; 2d rad
+          "10.5067/6UX3EDUNVUFK", "10.5067/6UX3EDUNVUFK",  \
+          "10.5067/6UX3EDUNVUFK", "10.5067/6UX3EDUNVUFK",  \
+          "10.5067/6UX3EDUNVUFK",                          \
+          "10.5067/IYDN3LNZ63UE", "10.5067/IYDN3LNZ63UE"/)    ; 2d chm
+
+  ; MIP
+  MIP = "Amon"
+
+  ; Frequency
+  FREQ = "mon"
+
+  ; CMOR table
+  base = getenv("cmor_tables")
+
+  CMOR_TABLE = getenv("cmor_tables") + \
+    (/"/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \  ; 3d asm
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP,                                \
+      "/cmip5/Tables/CMIP5_" + MIP, "/custom/CMOR_lwp.dat",        \  ; 2d int
+      "/cmip5/Tables/CMIP5_" + MIP,                                \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \  ; 2d rad
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP,  \
+      "/cmip5/Tables/CMIP5_" + MIP,                                \
+      "/cmip5/Tables/CMIP5_" + MIP, "/cmip5/Tables/CMIP5_" + MIP/)    ; 2d chm
+
+  ; Type
+  TYPE = "reanaly"
+
+  ; Version
+  VERSION = "5.2.0"
+
+  ; Global attributes
+  SOURCE = "https://goldsmr3.gesdisc.eosdis.nasa.gov/data/MERRA_MONTHLY/"
+  REF = "doi: "
+  COMMENT = "Goddard Earth Sciences Data and Information Services Center " + \
+            "(GES DISC)"
+
+end
+
+begin
+
+  ; Loop over variables
+  do vv = 0, dimsizes(VAR) - 1
+
+    log_info("Processing " + VAR(vv) + " (" + MIP + ")")
+    if (isvar("output")) then
+      delete(output)
+    end if
+
+    do yy = YEAR1, YEAR2
+      do mm = 1, 12
+
+        datestr = tostring(yy)
+        if (mm .lt. 10) then
+          datestr = datestr + "0"
+        end if
+        datestr = datestr + tostring(mm)
+
+        fname = systemfunc("ls " + input_dir_path + "MERRA???.prod.assim." + \
+                           SOURCEFILE(vv) + datestr + ".hdf")
```

Comment:
I see the file extension is harwored here, but the downloaded accepts options for nc or nc4 as well, is there a file handler anywhere else that is converting/extracting those to HDF5 or MERRA will never come in as netCDF?

Reply:
The original MERRA data are only available in .hdf format. Unfortunately... Since the stuff is archived, I do not expect this to change... But I am happy to change ".hdf" to ".*" if you like that better.

Reply:
aha gotcha! But then I'd not turn on the download of any nc or nc4 files at download point (via cmd line option) since those files will be ignored, if they do get downloaded at all - either that, or add an "*" option here, but in that case will the file handling have to change inside the NCL code ie to load and process nc vs HDF5? (sorry, I still don't speak any NCL :grin: )